pkg/webhooks/ingresscontroller/ingresscontroller.go

```diff
@@ -42,7 +54,22 @@ var (
 )
 
 type IngressControllerWebhook struct {
-	s runtime.Scheme
+	s          runtime.Scheme
+	kubeClient client.Client
+	// Allow caching install config and machineCIDR values...
+	machineCIDRIP  net.IP
+	machineCIDRNet *net.IPNet
+}
+
+// installConfig struct to load the min values needed from the install-config data.
+// Alternative would be to vendor all of github.com/openshift/installer/pkg/types to import the proper struct.
+// Required values: machineCidr info, anything else we gather from this? hcp vs classic, privatelink info, etc?
+type installConfig struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+	Networking        struct {
+		MachineCIDR string `json:"machineCIDR"`
+	} `json:"networking"`
```

Comment:
Personally, I think we should just import the proper struct from `github.com/openshift/installer/pkg/types` - it doesn't seem like the package is all that large and it'll potentially simplify a lot of logic under the `getMachineCIDR()` and `getClusterConfig()` functions

Reply:
Thanks, will do.

Reply:
Updated in latest commit.